web/src/app/chat/modal/SetDefaultModelModal.tsx

```diff
@@ -0,0 +1,174 @@
+import { usePopup } from "@/components/admin/connectors/Popup";
+
+export function SetDefaultModelModal({
+  llmProviders,
+  onClose,
+  setLlmOverride,
+  defaultModel,
+}: {
+  llmProviders: LLMProviderDescriptor[];
+  setLlmOverride: Dispatch<SetStateAction<LlmOverride>>;
+  onClose: () => void;
+  defaultModel: string | null;
+}) {
+  const { popup, setPopup } = usePopup();
+
+  const defaultModelDestructured = defaultModel
+    ? destructureValue(defaultModel)
+    : null;
+  const modelOptionsByProvider = new Map<
+    string,
+    { name: string; value: string }[]
+  >();
+  llmProviders.forEach((llmProvider) => {
+    const providerOptions = llmProvider.model_names.map(
+      (modelName: string) => ({
+        name: getDisplayNameForModel(modelName),
+        value: modelName,
+      })
+    );
+    modelOptionsByProvider.set(llmProvider.name, providerOptions);
+  });
+
+  const llmOptionsByProvider: {
+    [provider: string]: { name: string; value: string }[];
+  } = {};
+  const uniqueModelNames = new Set<string>();
+
+  llmProviders.forEach((llmProvider) => {
+    if (!llmOptionsByProvider[llmProvider.provider]) {
+      llmOptionsByProvider[llmProvider.provider] = [];
+    }
+
+    (llmProvider.display_model_names || llmProvider.model_names).forEach(
+      (modelName) => {
+        if (!uniqueModelNames.has(modelName)) {
+          uniqueModelNames.add(modelName);
+          llmOptionsByProvider[llmProvider.provider].push({
+            name: modelName,
+            value: structureValue(
+              llmProvider.name,
+              llmProvider.provider,
+              modelName
+            ),
+          });
+        }
+      }
+    );
+  });
+
+  const llmOptions = Object.entries(llmOptionsByProvider).flatMap(
+    ([provider, options]) => [...options]
+  );
+
+  const router = useRouter();
+  const handleChangedefaultModel = async (defaultModel: string | null) => {
+    try {
+      const response = await setUserDefaultModel(defaultModel);
+
+      if (response.ok) {
+        if (defaultModel) {
+          setLlmOverride(destructureValue(defaultModel));
+        }
+        setPopup({
+          message: "Default model updated successfully",
+          type: "success",
+        });
+        router.refresh();
+      } else {
+        throw new Error("Failed to update default model");
+      }
+    } catch (error) {
+      setPopup({
+        message: "Failed to update default model",
+        type: "error",
+      });
+    }
+  };
+
+  return (
+    <ModalWrapper
+      onClose={onClose}
+      modalClassName="rounded-lg bg-white max-w-xl"
+    >
+      <>
+        {popup}
+        <div className="flex mb-4">
+          <h2 className="text-2xl text-emphasis font-bold flex my-auto">
+            Set Default Model
+          </h2>
+        </div>
+
+        <Text className="mb-4">
+          Select a default Large Language Model (Generative AI model) to power
+          this Assistant.
```

Comment:
huh, I swear I added this comment but it seems I forgot to click confirm or something 😅 

This probably shouldn't be `this Assistant` since it applies to many assistants? Maybe something like "Select a default Large Language Model (Generative AI model) to power <Danswer | Whitelabeled Name>".

Additionally, it may be worth thinking about the best way to avoid confusion when the user selects a default model, but then they use an assistant with an overridden default 🤔